### ISSUE TITLE
3.0.0 fails to install on OS X

### ISSUE THREAD

**Issue by jamesspicer1980** [ibmdb/ruby-ibmdb#51]
I've run into issues installing this gem on my machine (El Capitan). After reading through other issues, I've resolved a few issue, but still have more. [Documentation](https://github.com/ibmdb/ruby-ibmdb/issues/50) would help!

1) IBM_DB_HOME is not set.

After installing db2_v101_macos_expc & macos64_odbc_cli and setting IBM_DB_HOME to either(??) install location, I was able to get to the next issue.

2) I guess installing the packages in part 1 wasn't the right move.

```
/opt/rubies/2.2.2/bin/ruby -r ./siteconf20160104-18336-8cphfx.rb extconf.rb
Detected 64-bit Ruby
 
Detected usage of IBM Data Server Driver package. Ensure you have downloaded 64-bit package of IBM_Data_Server_Driver and retry the 'gem install ibm_db' command
 
creating gil_release_version
creating unicode_support_version
checking for SQLConnect() in -ldb2... yes
checking for gil_release_version... yes
checking for unicode_support_version... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling ibm_db.c
ibm_db.c:106:9: warning: 'STRCASECMP' macro redefined [-Wmacro-redefined]
#define STRCASECMP strcasecmp
        ^
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:1801:9: note: previous definition is here
#define STRCASECMP(s1, s2) (st_locale_insensitive_strcasecmp((s1), (s2)))
        ^
ibm_db.c:698:12: warning: implicit declaration of function 'rb_thread_call_without_gvl' is invalid in C99 [-Wimplicit-function-declaration]
                                return rb_thread_call_without_gvl(func, data1, ubf, data2);                        
                                       ^
ibm_db.c:2360:41: warning: cast to 'SQLPOINTER' (aka 'void *') from smaller integer type 'SQLINTEGER' (aka 'int') [-Wint-to-void-pointer-cast]
        handleAttr_args->valuePtr    =  (SQLPOINTER)(enable_numeric_literals);
                                        ^
ibm_db.c:2304:32: warning: 'memset' call operates on objects of type 'SQLWCHAR' (aka 'unsigned short') while the size is based on a different type 'SQLWCHAR *' (aka 'unsigned short *') [-Wsizeof-pointer-memaccess]
      memset(server, 0, sizeof(server));
             ~~~~~~            ^~~~~~
ibm_db.c:2304:32: note: did you mean to dereference the argument to 'sizeof' (and multiply it by the number of elements)?
      memset(server, 0, sizeof(server));
                               ^~~~~~
ibm_db.c:2433:1: warning: control may reach end of non-void function [-Wreturn-type]
}
^
ibm_db.c:2538:26: warning: comparison between pointer and integer ('VALUE' (aka 'unsigned long') and 'void *')
                if(helper_args-> entry == NULL)
                   ~~~~~~~~~~~~~~~~~~~ ^  ~~~~
ibm_db.c:3328:37: warning: cast to 'SQLPOINTER' (aka 'void *') from smaller integer type 'SQLINTEGER' (aka 'int') [-Wint-to-void-pointer-cast]
        handleAttr_args->valuePtr = (SQLPOINTER)autocommit;
                                    ^
ibm_db.c:5676:18: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
        rb_warn( RSTRING_PTR(error) );
                 ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:5816:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
          rb_warn( RSTRING_PTR(error) );
                   ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:7027:16: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
      rb_warn( RSTRING_PTR(error) );
               ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:9284:14: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
    rb_warn( RSTRING_PTR(error) );
             ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:10574:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
          rb_warn( RSTRING_PTR(error) );
                   ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:10584:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
          rb_warn( RSTRING_PTR(error) );
                   ^~~~~~~~~~~~~~~~~~
/opt/rubies/2.2.2/include/ruby-2.2.0/ruby/ruby.h:866:5: note: expanded from macro 'RSTRING_PTR'
    (!(RBASIC(str)->flags & RSTRING_NOEMBED) ? \
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ibm_db.c:11252:6: warning: comparison of address of 'connection' equal to a null pointer is always false [-Wtautological-pointer-compare]
        if(&connection == NULL)
            ^~~~~~~~~~    ~~~~
14 warnings generated.
compiling ruby_ibm_db_cli.c
linking shared-object ibm_db.bundle
ld: unknown option: -R$ORIGIN/clidriver/lib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [ibm_db.bundle] Error 1

make failed, exit code 2
```

3) Can't find the installer for IBM Data Server Driver package

I can find a lot of links for Linux and Windows, but nothing for OS X. What's up? http://www-01.ibm.com/support/docview.wss?uid=swg21385217

**Comment by arvindgu** [comment]
Today, we have released *new* version of ibm_db gem **3.0.1** to fix the installation problem in El Capitan. Please try this and let us know. 

You do not need IBM Data Server Driver package, as we have already simplified installation process to auto-download the required IBM Data Server Driver package when ibm_db gem is being installed.
We will further work to improve documentation.

**Comment by jonstorer** [comment]
@arvindgu thanks! that helped!
